tests/scripts/all.sh

```diff
@@ -793,6 +794,98 @@ pre_generate_files() {
     fi
 }
 
+is_cc_target_aarch64() {
+    local cc=${1:-cc}
+    local target=$(${cc} -dumpmachine)
+    case ${target} in
+        *aarch64*) true;;
+        *) false;;
+    esac
+}
+
+is_cc_support_sha512_a64() {
+    local cc=${1:-cc}
+
+    # SHA512 instruction is for aarch64
+    is_cc_target_aarch64 "${cc}" || return 1
+
+    local major
+    major=$(${cc} -dumpversion | cut -d '.' -f 1)
+
+    # Check the compiler's version depending on its type (GCC/Clang)
+    case $(${cc} --version 2>&1) in
+        *gcc*|*cc*)
+            if [ "${major}" -lt "8" ]; then
+                false
+            else
+                true
+            fi
+            ;;
+        *clang*)
+            if [ "${major}" -lt "7" ]; then
+                false
+            else
+                true
+            fi
+            ;;
+        *) false;;
+    esac
+}
+
+check_cc_version() {
+    local component=$1
+
+    # These components don't need to be checked:
+    #   - build_arm{cc|_xxx} don't use gcc/clang
+    #   - build_module_alt and test_psa_crypto_config_accel_hash_use_psa
+    #     enable full config but unset MBEDTLS_SHA{512/256}_USE_A64_CRYPTO_*
+    if [[ "$component" =~ ^build_arm.*$ ]] || \
+       [[ "$component" == "build_module_alt" ]] || \
+       [[ "$component" == "test_psa_crypto_config_accel_hash_use_psa" ]]; then
+        return 0
+    fi
```

Comment:
I do not think we should filter them by name. 

The condition should be ：
- `MBEDTLS_SHA{512/256}_USE_A64_CRYPTO_*` is enabled
- compiler target is `aarch64`